File: Archive#Unreleased/Perform String Shifts/leetcode.rb
# @param {String} s
# @param {Integer[][]} shift
# @return {String}
def string_shift(s, shift)
  sz = s.length()
  movement = shift.map {|e| e[0] == 1 ? e[1] : -e[1] }.inject(:+) % sz
  return s if movement == 0
  s[-movement..-1] + s[0...-movement]
end

File: Archive#Unreleased/Perform String Shifts/leetcode.py
class Solution:
    def stringShift(self, s: str, shift: List[List[int]]) -> str:
        movement = sum((d * 2 - 1) * v for d, v in shift) % len(s)
        if movement == 0:
            return s
        else:
            return s[-movement:] + s[:-movement]
